src/state/opamSysInteract.ml

```diff
@@ -596,6 +598,108 @@ let packages_status ?(env=OpamVariable.Map.empty) packages =
       |> package_set_of_pkgpath
     in
     compute_sets sys_installed
+  | Nixos ->
+      let nix_env = "nix-env" in
+      let is_package_installed package =
+        (* takes a single package and checks whether it is installed *)
+        let pname = OpamSysPkg.to_string package in
+        let args = ["--query"; pname] in
+        match run_command nix_env args with
+        | (0, _) -> true
+        | (1, _) -> false
+        | (code, output) ->
+          Printf.ksprintf failwith
+            "failed with unexpected exit code %d running command:\n    %s\n\nOutput:\n%s"
+            code (String.concat " " (nix_env::args)) (String.concat "\n" output)
+      in
+      let all_pnames_of_nix_package_json : OpamJson.t -> string list option = function
```

Comment:
Wouldn't be faster to directly parse output (list filtering / regexp) than work on parsed json output ? see `Arch` for example.